core/storage/blog/gae_models.py

```diff
@@ -334,15 +334,62 @@ def has_reference_to_user_id(cls, user_id):
             cls.editor_ids == user_id).get(keys_only=True) is not None
 
     @classmethod
-    def get_by_user(cls, user_id):
-        """Retrieves the rights object for all blog posts assigned to given user
+    def get_published_models_by_user(cls, user_id, limit=None):
+        """Retrieves the blog post rights objects for published blog posts for
+        which the given user is an editor.
 
         Args:
-            user_id: str. ID of user.
+            user_id: str. ID of the author of the blog post.
+            limit: int|None. The maximum number of BlogPostRightsModels to be
+                fetched. If None, all existing published models by user will be
+                fetched.
 
         Returns:
             list(BlogPostRightsModel). The list of BlogPostRightsModel objects
-            in which the given user is a editor.
+            in which the given user is an editor. The list will be ordered
+            according to the time when the model was last updated.
+        """
+        query = cls.query(
+            cls.editor_ids == user_id, cls.blog_post_is_published == True # pylint: disable=singleton-comparison
+        ).order(-cls.last_updated)
+        if limit is not None:
```

Comment:
(Optional) Can replace these three lines with `return query.fetch(limit) if limit is not None else query.fetch()`

Reply:
Done.